spring-integration-core/src/main/java/org/springframework/integration/router/ExpressionEvaluatingRouter.java

```diff
@@ -35,4 +35,9 @@ public ExpressionEvaluatingRouter(Expression expression) {
 		setPrimaryExpression(expression);
 	}
 
+	public ExpressionEvaluatingRouter(String expressionString) {
+		super(new ExpressionEvaluatingMessageProcessor<Object>(EXPRESSION_PARSER.parseExpression(expressionString)));
+		setPrimaryExpression(getExpression());
```

Comment:
Why `this(EXPRESSION_PARSER.parseExpression(expressionString))` doesn't work?

And JavaDocs, please